app/controllers/api/v1/notes_controller.rb

```diff
@@ -0,0 +1,45 @@
+module Api
+  module V1
+    class NotesController < ApplicationController
+      def index
+        type = params[:type]
+        page_size = params[:page_size]
+        max_page_size = 100
+
+        if page_size.to_i > max_page_size
+          return render json: { error: "page_size is too long, max allowed is #{max_page_size}" }, status: :bad_request
+        end
+
+        if type && !Note.types.keys.include?(type)
+          return render json: { error: "invalid type #{type}" }, status: :unprocessable_entity
+        end
+        render json: notes_filtered, status: :ok, each_serializer: IndexNoteSerializer
```

Comment:
```suggestion
        render json: notes, status: :ok, each_serializer: IndexNoteSerializer
```
A esto solemos llamarlo notes directamente (el plural del recurso)

Reply:
Perfecto! Queda mejor.